grails-app/controllers/org/broadinstitute/orsp/webservice/SampleCollectionController.groovy

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.orsp.webservice
+
+import com.google.gson.Gson
+import grails.converters.JSON
+import grails.rest.Resource
+import grails.rest.RestfulController
+import groovy.util.logging.Slf4j
+import org.broadinstitute.orsp.SampleCollection
+import org.springframework.web.multipart.MultipartFile
+
+import groovyx.net.http.HttpBuilder
+import org.broadinstitute.orsp.webservice.BspCollection
+import org.broadinstitute.orsp.config.SampleCollectionConfiguration
+
+@Slf4j
+@Resource(readOnly = false, formats = ['JSON', 'CSV'])
+class SampleCollectionController extends RestfulController<SampleCollection> {
+
+    Gson gson = new Gson()
+    SampleCollectionConfiguration sampleCollectionConfiguration
+
+    SampleCollectionController() {
+       super(SampleCollection)
+    }
+
+    def index() {
+        def json = gson.toJson("Unimplemented")
+        render(text: json, contentType: "application/json")
+    }
+
+    /**
+     * Receives a csv file containing all BSP Sample Collections 
+     * from a cron job running in Broad's internal network, 
+     * finds out new sample collection (not in ORSP db)
+     * and save it on ORSP db.
+     *
+     * @return List of added sample collections
+     */
+    def save() {
+
+        if (sampleCollectionConfiguration.token != request.getHeader('TOKEN')) {
+            log.info("Unauthorized BSP Sync request at: ${new Date().getDateTimeString()}")
```

Comment:
I think we should be logging all requests for safety, and log as much of the request as possible in the failure case. We will need to follow up on those and will need as much info as possible to diagnose the offending requestor.